smalltalksrc/VMMaker/CoInterpreter.class.st

```diff
@@ -2604,6 +2606,28 @@ CoInterpreter >> executeFullCogBlock: cogMethod closure: closure mayContextSwitc
 	"NOTREACHED"
 ]
 
+{ #category : #'message sending' }
+CoInterpreter >> executeNewMethod: eagerlyCompile [
+	"if not primitive, or primitive failed, activate the method"
+	<inline: true>
+	| inInterpreter |	
+	inInterpreter := self isInstructionPointerInInterpreter:
+		                 instructionPointer.
+	self executePrimitiveFromInterpreter: inInterpreter ifFail: [
+		| methodHeader |
+		eagerlyCompile ifTrue: [
+			"Eagerly compile it if appropriate so that doits are fast."
+			methodHeader := self rawHeaderOf: newMethod.
+			(self isCogMethodReference: methodHeader) ifFalse: [
+				(self methodWithHeaderShouldBeCogged: methodHeader)
+					ifTrue: [
+						cogit cog: newMethod selector: objectMemory nilObject.
+						methodHeader := self rawHeaderOf: newMethod ]
+					ifFalse: [ self maybeFlagMethodAsInterpreted: newMethod ] ]].
+		"if not primitive, or primitive failed, activate the method"
+		self activateNewMethod ]
+]
+
```

Comment:
This is the major change in here, now this method can be parametrised to eagerly compile methods (or not). By default the interpreter will not do it.